File: src/main/java/no/nav/pensjon/innsyn/domain/FppAfp.java
package no.nav.pensjon.innsyn.domain;

/**
 * FPP = Framtidige pensjonspoeng
 * AFP = Avtalefestet pensjon
 */
public class FppAfp {

    private final String status;
    private final double fppAfp;
    private final String gjelderFom;
    private final String gjelderTom;
    private final double afpPensjonsgrad;
    private final String afpType;

    public FppAfp(String status,
                  double fppAfp,
                  String gjelderFom,
                  String gjelderTom,
                  double afpPensjonsgrad,
                  String afpType) {
        this.status = status;
        this.fppAfp = fppAfp;
        this.gjelderFom = gjelderFom;
        this.gjelderTom = gjelderTom;
        this.afpPensjonsgrad = afpPensjonsgrad;
        this.afpType = afpType;
    }

    public String getStatus() {
        return status;
    }

    public double getFppAfp() {
        return fppAfp;
    }

    public String getGjelderFom() {
        return gjelderFom;
    }

    public String getGjelderTom() {
        return gjelderTom;
    }

    public double getAfpPensjonsgrad() {
        return afpPensjonsgrad;
    }

    public String getAfpType() {
        return afpType;
    }
}

File: src/main/kotlin/no/nav/pensjon/innsyn/domain/popp/FppAfp.kt
package no.nav.pensjon.innsyn.domain.popp

import no.nav.pensjon.innsyn.domain.Domain
import no.nav.pensjon.innsyn.domain.popp.support.FppAfpStatus
import java.time.LocalDate
import javax.persistence.*

/**
 * FPP = Framtidige pensjonspoeng
 * AFP = Avtalefestet pensjon
 */
@Entity
@Table(name = "T_FPP_AFP")
@SecondaryTable(name = "T_K_FPP_AFP_S", pkJoinColumns = [PrimaryKeyJoinColumn(name = "K_FPP_AFP_S")])
data class FppAfp(
        @Id
        @Column(name = "PERSON_ID")
        private val personId: Int,
        @ManyToOne
        @JoinColumn(name = "K_FPP_AFP_S")
        private val fppAfpStatus: FppAfpStatus,
        @Column(name = "AFP_FPP")
        val fppAfp: Double,
        @Column(name = "VIRK_FOM")
        val gjelderFom: LocalDate,
        @Column(name = "VIRK_TOM")
        val gjelderTom: LocalDate,
        @Column(name = "AFP_PENSJONSGRAD")
        val afpPensjonsgrad: Double,
        @Column(name = "AFP_TYPE")
        val afpType: String
): Domain {
    @Transient
    val status = fppAfpStatus.dekode
    @Transient
    override val fields = setOf(::status, ::fppAfp, ::gjelderFom, ::gjelderTom, ::afpPensjonsgrad, ::afpType)
}